rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,273 @@
+
+        agent = results_object.get("data").get("assets")[0].get("agent")
+        quarantine_state = agent.get("quarantineState").get("currentState")
+        agent_status = agent.get("agentStatus")
+
+        is_online = True if agent_status == "ONLINE" else False
+        is_quarantine_requested = True if quarantine_state == "QUARANTINE_IN_PROGRESS" else False
+        is_unquarantine_requested = True if quarantine_state == "UNQUARANTINE_IN_PROGRESS" else False
+        is_is_quarantined = True if (quarantine_state == "QUARANTINED" or quarantine_state == "UNQUARANTINE_IN_PROGRESS") else False
+
+        return {
+            "is_currently_quarantined": is_is_quarantined,
+            "is_asset_online": is_online,
+            "is_quarantine_requested": is_quarantine_requested,
+            "is_unquarantine_requested": is_unquarantine_requested
+        }
+
+    def connection_test(self):
+        # Return the first org to verify the connection works
+        graph_ql_payload = {
+            "query":"{ organizations(first: 1) { edges { node { id } } totalCount } }"
+        }
+
+        # If no exceptions are thrown, we have a valid connection
+        self._post_payload(graph_ql_payload)
+        return True
+
+    #################
+    # Private Methods
+    #################
+
+    def _get_headers(self):
+        """
+        This build and returns headers for the request session
+
+        :return: dict
+        """
+        return {
+            "X-Api-key": self.api_key,
+            "Accept-Version": "kratos",
+            "Accept": "*/*",
+            "Accept-Encoding": "gzip, deflate, br"
+        }
+
+    def _post_payload(self, payload):
+        """
+        This will post a given payload to the API using the connection session
+
+        :param payload: dict
+        :return: dict
+        """
+        result = self.session.post(self.endpoint, json=payload)
+
+        try:
+            result.raise_for_status()
+        except:
+            raise APIException(cause="Error connecting to the Insight Agent API.",
+                               assistance="Please check your Organization ID and API key.\n",
+                               data=result.text)
+
+        results_object = result.json()
+
+        if results_object.get("errors"):
+            raise APIException(cause="The Insight Agent API returned errors",
+                               assistance=results_object.get("errors"))
+
+        return results_object
+
+    def _setup_endpoint(self, region_string):
+        region_code = region_map.get(region_string)
+
+        if region_code:
+            endpoint = f"https://{region_code}.api.insight.rapid7.com/graphql"
+        else:
+            # It's an enum, hopefully this never happens.
+            raise APIException(cause="Region not found.",
+                               assistance="Region code was not found for selected region. Available reagions are: United States, "
```

Comment:
Typo fix

```suggestion
                               assistance="The available regions are: United States, "
```

is this of type PluginException? if not, we should stick to that in the future we will parse those message in the UI.